text_recognizer/data/iam_synthetic_paragraphs.py

```diff
@@ -56,11 +58,14 @@ def prepare_data(self, *args, **kwargs) -> None:
     def setup(self, stage: str = None) -> None:
         rank_zero_info(f"IAMSyntheticParagraphs.setup({stage}): Loading trainval IAM paragraph regions and lines...")
 
-        if stage == "fit" or stage is None:
-            line_crops, line_labels = load_line_crops_and_labels("train", PROCESSED_DATA_DIRNAME)
+        def _load_dataset(split: str, transform: ParagraphStem) -> BaseDataset:
```

Comment:
Is this type annotation too strict? Do we really only allow `ParagraphStem`s here? I don't think we actually have a useful type annotation to provide here, because the type is something like "callable that takes a tensor and returns a tensor", which is insufficient to express actually useful things like shape assumptions. So my preference would be not to annotate.

Reply:
Fixed at https://github.com/full-stack-deep-learning/fsdl-text-recognizer-2022/pull/31/files#diff-cd205268606d2b02590586c359105225d4cffe942d0e76f9e4c50d33dd260e17R76